server/app/api/config/exception_handler.py

```diff
@@ -26,9 +27,9 @@ def register_exception_handler(app: FastAPI):
     def _UserException(r: Request, e: UserException):
         raise HTTPException(400, detail={"message": str(e)}) 
     
-    @app.exception_handler(ValidationError)
-    def _ValidationError(r: Request, e: ValidationError):
-        raise HTTPException(400, detail={"message": e.errors()}) 
+    @app.exception_handler(RequestValidationError)
+    def _ValidationError(r: Request, e: RequestValidationError):
+        raise HTTPException(400, detail={"message": str(e.errors())})
```

Comment:
Perhaps the user is receiving too many details about the error.

Reply:
Here's why I'm against this:


```py
class ErrorDetails(_TypedDict):
    type: str
    loc: tuple[int | str, ...]
    msg: str
    input: _Any
    ctx: _NotRequired[dict[str, _Any]]
```

When a Pydantic validation fails, we'll get a list of `ErrorDetails` in the raised exception. Currently my code returns the exception message as-is, which more or less converts the `ErrorDetails` objects into JSON.

What we would want is to have the exception handler convert that data into a human-readable message. I could create the message from `msg` and `loc` fields (`msg` is too vague, while `loc` provides us with which variable failed to validate). 

For example, when setting the new password, I would have to use: 

```json
loc=("body", "new_password")
msg="String should contain at least 8 characters"
```

So I'd have to:
1) take `loc[1]`
2) convert it into readable form
3) combine it with `msg` into a coherent sentence

All three of these tasks are non-trivial:
1) It may not be `loc[1]` per se. `loc` is a `tuple[int | str, ...]`, so we don't know.
2) This is okay for `new_password -> New password`, but it depends on how the variable is named on the server
3) One validation "type" = one sentence, minus the grammar

---

An alternative would be to create a custom Pydantic validator, but that's too much work as well. There's no way to modify the message for a valdiatior either (it's not like Spring Data, unfortunately).

Ultimately, I'd leave things as-is because this is a very insignificant thing. We have tons of features that need implementation.